external-crates/move/crates/move-compiler/src/sui_mode/linters/freezing_capability.rs

```diff
@@ -90,27 +90,26 @@ impl<'a> TypingVisitorContext for Context<'a> {
     }
 }
 
-impl<'a> Context<'a> {
-    fn is_freeze_function(&self, fun: &T::ModuleCall) -> bool {
-        FREEZE_FUNCTIONS.iter().any(|(addr, module, fname)| {
-            fun.module.value.is(*addr, *module) && &fun.name.value().as_str() == fname
-        })
-    }
-
-    fn check_type_arguments(&mut self, fun: &T::ModuleCall, loc: Loc) {
-        for sp!(_, type_arg) in &fun.type_arguments {
-            if let Some(sp!(_, TypeName_::ModuleType(_, struct_name))) = type_arg.type_name() {
-                if self.capability_regex.is_match(struct_name.value().as_str()) {
-                    self.report_freeze_capability(loc);
-                    break;
-                }
-            }
-        }
-    }
+fn is_freeze_function(fun: &T::ModuleCall) -> bool {
+    FREEZE_FUNCTIONS.iter().any(|(addr, module, fname)| {
+        fun.module.value.is(*addr, *module) && &fun.name.value().as_str() == fname
+    })
+}
 
-    fn report_freeze_capability(&mut self, loc: Loc) {
-        let msg = "Freezing a capability-like type can lead to design issues.";
-        let diag = diag!(FREEZE_CAPABILITY_DIAG, (loc, msg));
-        self.env.add_diag(diag);
+fn check_type_arguments(context: &mut Context, fun: &T::ModuleCall, loc: Loc) {
+    for sp!(_, type_arg) in &fun.type_arguments {
+        let Some(sp!(_, TypeName_::ModuleType(_, struct_name))) = type_arg.type_name() else {
+            continue;
+        };
+        if REGEX.is_match(struct_name.value().as_str()) {
+            let msg = format!(
+                "The type {} is potentially a capability based on its name. Freezing a capability might lock out critical operations or otherwise open access to operations that otherwise should be restricted.",
+                core::error_format_(type_arg, &core::Subst::empty()),
+            );
+            let diag = diag!(FREEZE_CAPABILITY_DIAG, (loc, msg));
+            context.env.add_diag(diag);
```

Comment:
More recently I have been trying to follow Rust and split up longer message like this into a location message and a note, e.g., :

```suggestion
            let msg = format!(
                "The type {} is potentially a capability based on its name",
                core::error_format_(type_arg, &core::Subst::empty()),
            );
            let mut diag = diag!(FREEZE_CAPABILITY_DIAG, (loc, msg));
            diag.add_note("Freezing a capability might lock out critical operations or otherwise open access to operations that otherwise should be restricted");
            context.env.add_diag(diag);
```

Reply:
Maybe. Keep in mind though the note might not be rendered in the IDE easily